src/CoreWCF.Primitives/src/CoreWCF/Security/NegotiationTokenAuthenticator.cs

```diff
@@ -0,0 +1,1056 @@
+
+        NegotiationTokenAuthenticatorStateCache<T> stateCache;
+        RenewedSecurityTokenHandler renewedSecurityTokenHandler;
+        NegotiationHost negotiationHost;
+        bool encryptStateInServiceToken;
+        TimeSpan serviceTokenLifetime;
+        int maximumCachedNegotiationState;
+        TimeSpan negotiationTimeout;
+        bool isClientAnonymous;
+        SecurityStandardsManager standardsManager;
+        SecurityAlgorithmSuite securityAlgorithmSuite;
+        SecurityTokenParameters issuedSecurityTokenParameters;
+        ISecurityContextSecurityTokenCache issuedTokenCache;
+        BindingContext issuerBindingContext;
+        Uri listenUri;
+        string sctUri;
+       // AuditLogLocation auditLogLocation;
+        bool suppressAuditFailure;
+       // AuditLevel messageAuthenticationAuditLevel;
+        SecurityStateEncoder securityStateEncoder;
+        SecurityContextCookieSerializer cookieSerializer;
+        IMessageFilterTable<EndpointAddress> endpointFilterTable;
+        IssuedSecurityTokenHandler issuedSecurityTokenHandler;
+        int maxMessageSize;
+        IList<Type> knownTypes;
+        int maximumConcurrentNegotiations;
+        List<IChannel> activeNegotiationChannels1;
+        List<IChannel> activeNegotiationChannels2;
+        IOThreadTimer idlingNegotiationSessionTimer;
+        bool isTimerCancelled;
+
+        protected NegotiationTokenAuthenticator() : base()
+        {
+            InitializeDefaults();
+        }
+
+        public IssuedSecurityTokenHandler IssuedSecurityTokenHandler
+        {
+            get
+            {
+                return this.issuedSecurityTokenHandler;
+            }
+            set
+            {
+                this.issuedSecurityTokenHandler = value;
+            }
+
+        }
+
+        public RenewedSecurityTokenHandler RenewedSecurityTokenHandler
+        {
+            get
+            {
+                return this.renewedSecurityTokenHandler;
+            }
+            set
+            {
+                this.renewedSecurityTokenHandler = value;
+            }
+        }
+
+        // settings
+        public bool EncryptStateInServiceToken
+        {
+            get
+            {
+                return this.encryptStateInServiceToken;
+            }
+            set
+            {
+                this.CommunicationObject.ThrowIfDisposedOrImmutable();
+                this.encryptStateInServiceToken = value;
+            }
+        }
+
+        public TimeSpan ServiceTokenLifetime
+        {
+            get
+            {
+                return this.serviceTokenLifetime;
+            }
+            set
+            {
+                this.CommunicationObject.ThrowIfDisposedOrImmutable();
+                if (value <= TimeSpan.Zero)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException("value", "TimeSpanMustbeGreaterThanTimeSpanZero"));
+                }
```

Comment:
nameof(value), and the second parameter should be a string resource and not an constant string.

Reply:
done